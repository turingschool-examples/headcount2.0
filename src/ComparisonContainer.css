
.comp-container {
  padding-top: 2rem;
  background-color: #C2B2B4;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.comp-card {
  padding: .5rem;
  border: 2px solid chartreuse;
  height: 20rem;
  width: 15vw;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  background-color: #ffffff;
  border-radius: .9rem;
}

.total-card {
  padding: .5rem;
  height: 15rem;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  border: 2px solid chartreuse;
  width: 15vw;
  background-color: #ffffff;

}