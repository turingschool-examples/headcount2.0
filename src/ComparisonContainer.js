import React from 'react';
import Card from './Card';

const ComparisonContainer = () => {

  return(
    <div>
      <h3>Select two districts to compare, they will be displayed here.</h3>
    </div>
  )
}

export default ComparisonContainer;