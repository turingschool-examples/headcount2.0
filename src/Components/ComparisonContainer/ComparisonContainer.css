.comparison-container {
  display: flex;
  justify-content: space-evenly;
  flex-direction: row;
  flex-wrap: wrap;
  max-width: 1200px;
  margin: auto
}

.comparison-card {
  font-family: avenir, sans-serif;
  font-size: 1em;
  text-align: center;
}

.district-average {
  font-size: 1.5em;
}

.comparison-figure {
  font-size: 3em;
  color: red;
}