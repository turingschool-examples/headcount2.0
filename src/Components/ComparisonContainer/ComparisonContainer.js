import React from 'react';
import DistrictCard from '../DistrictCard/DistrictCard'
import PropTypes from 'prop-types';


const ComparisonContainer = ({ comparedDistricts }) => {
  const districtNames = Object.keys(comparedDistricts)
  const districtCards = districtNames.map(district => {
    return <DistrictCard 
              district={comparedDistricts[district]} 
            />
  })

  return (
    <div className="comparison-container">
      { districtCards }
    </div>
  )
} 


ComparisonContainer.propTypes = {
  comparedDistricts: PropTypes.object.isRequired
}

export default ComparisonContainer;