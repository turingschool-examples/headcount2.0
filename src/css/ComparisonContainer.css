.comparison-container {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
}

@media only screen and (max-width: 1350px) {
  .comparison-container {
    grid-template-columns: repeat(3, 1fr) !important;
  }
}
