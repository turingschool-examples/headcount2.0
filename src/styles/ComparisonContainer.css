.comparison-cards {
  width:100%;
  height: 575px;
  background-color: lightblue;
  text-align: center;
}