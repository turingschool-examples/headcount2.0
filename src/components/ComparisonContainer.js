import React, { Component } from 'react';
import Card from './Card';
import '../styles/ComparisonContainer.css';

const ComparisonContainer = (props) => {

  const dataArray = Object.keys(props.cardCompData)
  const newCard = dataArray.map((location, index) => <Card data={ props.cardCompData[location]} key={index}/>)

  return (
    <section className='comparison-cards'>
      <h1>Comparison Section</h1>
      {newCard}
    </section>
  )
}

export default ComparisonContainer