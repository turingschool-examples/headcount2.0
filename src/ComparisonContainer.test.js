import React from 'react';
import ComparisonContainer from './ComparisonContainer';
import { shallow } from 'enzyme';

describe('ComparisonContainer', () => {

  it('matches the snapshot', () => {
    const wrapper = shallow(<ComparisonContainer />);
    expect(wrapper).toMatchSnapshot()
  });

})