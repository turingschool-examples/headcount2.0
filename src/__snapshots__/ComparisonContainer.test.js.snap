// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3>
        Select two districts to compare, they will be displayed here.
</h3>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Select two districts to compare, they will be displayed here.",
      },
      "ref": null,
      "rendered": "Select two districts to compare, they will be displayed here.",
      "type": "h3",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3>
          Select two districts to compare, they will be displayed here.
</h3>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select two districts to compare, they will be displayed here.",
        },
        "ref": null,
        "rendered": "Select two districts to compare, they will be displayed here.",
        "type": "h3",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
